xds/src/test/java/io/grpc/xds/LocalityStoreTest.java

```diff
@@ -235,28 +240,61 @@ public void updateLocalityStore_interceptPickResultUponPickReadySubchannel() {
     childHelpers.get("sz1").updateBalancingState(READY, subchannelPicker1);
     childHelpers.get("sz2").updateBalancingState(READY, subchannelPicker2);
 
     assertThat(pickerFactory.totalReadyLocalities).isEqualTo(2);
     ArgumentCaptor<SubchannelPicker> interLocalityPickerCaptor = ArgumentCaptor.forClass(null);
     verify(helper, times(2)).updateBalancingState(eq(READY), interLocalityPickerCaptor.capture());
     SubchannelPicker interLocalityPicker = interLocalityPickerCaptor.getValue();
-    for (int i = 0; i < pickerFactory.totalReadyLocalities; i++) {
-      pickerFactory.nextIndex = i;
-      interLocalityPicker.pickSubchannel(pickSubchannelArgs);
-    }
-    verify(statsStore).interceptPickResult(same(result1), eq(locality1));
-    verify(statsStore).interceptPickResult(same(result2), eq(locality2));
+
+    pickerFactory.nextIndex = 0;
+    PickResult pickResult1 = interLocalityPicker.pickSubchannel(pickSubchannelArgs);
+    ArgumentCaptor<OrcaPerRequestReportListener> listenerCaptor1 = ArgumentCaptor.forClass(null);
+    verify(orcaPerRequestUtil)
+        .newOrcaClientStreamTracerFactory(any(ClientStreamTracer.Factory.class),
+            listenerCaptor1.capture());
+    assertThat(listenerCaptor1.getValue()).isInstanceOf(MetricsRecordingListener.class);
+    MetricsRecordingListener listener1 = (MetricsRecordingListener) listenerCaptor1.getValue();
+    assertThat(listener1.getCounter())
+        .isSameInstanceAs(fakeLoadStatsStore.localityCounters.get(locality1));
+    assertThat(pickResult1.getStreamTracerFactory()).isInstanceOf(ClientLoadRecorder.class);
+    ClientLoadRecorder recorder1 = (ClientLoadRecorder) pickResult1.getStreamTracerFactory();
+    assertThat(recorder1.getCounter())
+        .isSameInstanceAs(fakeLoadStatsStore.localityCounters.get(locality1));
+    assertThat(recorder1.delegate()).isSameInstanceAs(metricsRecorder1);
+
+    pickerFactory.nextIndex = 1;
+    PickResult pickResult2 = interLocalityPicker.pickSubchannel(pickSubchannelArgs);
+    ArgumentCaptor<OrcaPerRequestReportListener> listenerCaptor2 = ArgumentCaptor.forClass(null);
+    verify(orcaPerRequestUtil, times(2))
+        .newOrcaClientStreamTracerFactory(any(ClientStreamTracer.Factory.class),
+            listenerCaptor2.capture());
+    assertThat(listenerCaptor2.getValue()).isInstanceOf(MetricsRecordingListener.class);
+    MetricsRecordingListener listener2 = (MetricsRecordingListener) listenerCaptor2.getValue();
+    assertThat(listener2.getCounter())
+        .isSameInstanceAs(fakeLoadStatsStore.localityCounters.get(locality2));
+    assertThat(pickResult2.getStreamTracerFactory()).isInstanceOf(ClientLoadRecorder.class);
+    ClientLoadRecorder recorder2 = (ClientLoadRecorder) pickResult2.getStreamTracerFactory();
+    assertThat(recorder2.getCounter())
+        .isSameInstanceAs(fakeLoadStatsStore.localityCounters.get(locality2));
+    assertThat(recorder2.delegate()).isSameInstanceAs(metricsRecorder2);
```

Comment:
This is quite a few lines to repeat. Have you considered making a loop?

Reply:
Done.